src/services/summarize-state/index.js

```diff
@@ -0,0 +1,80 @@
+const _ = require('lodash');
+const hooks = require('./hooks');
+
+const pinStates = require('../../constants/pin-states');
+
+// Models
+const Department = require('../department/department-model');
+const Organization = require('../organization/organization-model');
+const Pin = require('../pin/pin-model');
+
+class Service {
+  constructor(options) {
+    this.options = options || {};
+  }
+
+  find(params) {
+    const startDate = params.query.start_date;
+    const endDate = params.query.end_date;
+    const organizationName = params.query.organization;
+    if (!startDate || !endDate) {
+      return Promise.reject(new Error('Please specified both start_date and end_date'));
+    }
+    // Find organization id
+    return Organization.findOne({ name: organizationName })
+      .then(oneOrg => {
+        if (!oneOrg) {
+          return Promise.reject(new Error(`Not found ${organizationName} in database`));
+        }
+        // Put params into a mongoose query format.
+        const query = {
+          organization: oneOrg._id,  // eslint-disable-line no-underscore-dangle
+          created_time: {
+            $gte: startDate,
+            $lte: endDate,
+          },
+        };
+        // Find pins that belong to the organization
+        return Pin.find(query).select('status assigned_department').limit(10000);
```

Comment:
Why should we limit? It can cause wrong result in the dashboard and KPI. (Anyway, in reality, it might not reach 10,000).